collector/aws/elasticsearch.go

```diff
@@ -0,0 +1,290 @@
+				EBSPrice, err = esm.pricingClient.GetPrice(esm.GetPricingFilterInput([]*pricing.Filter{
+					{
+						Type:  awsClient.String("TERM_MATCH"),
+						Field: awsClient.String("storageMedia"),
+						Value: awsClient.String(storageMedia),
+					},
+				}), "", esm.region)
+				if err != nil {
+					log.WithError(err).Error("Could not get ebs price")
+					continue
+				}
+				hourlyEBSVolumePrice = (EBSPrice * float64(*cluster.EBSOptions.VolumeSize)) / collector.TotalMonthHours
+			}
+		}
+
+		log.WithFields(log.Fields{
+			"instance_hour_price": instancePrice,
+			"ebs_hour_price":      hourlyEBSVolumePrice,
+			"region":              esm.region}).Debug("Found the following price list")
+
+		callerIdentityOutput, err := esm.stsClient.client.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+		if err != nil {
+			log.WithError(err).Error("Could not get AWS caller identity to get account id")
+			continue
+		}
+
+		for _, metric := range esm.metrics {
+			log.WithFields(log.Fields{
+				"cluster_arn": *cluster.ARN,
+				"metric_name": metric.Description,
+			}).Debug("checking metric")
+
+			period := int64(metric.Period.Seconds())
+			metricEndTime := now.Add(time.Duration(-metric.StartTime))
+			metricInput := cloudwatch.GetMetricStatisticsInput{
+				Namespace:  &esm.namespace,
+				MetricName: &metric.Description,
+				Period:     &period,
+				StartTime:  &metricEndTime,
+				EndTime:    &now,
+				Dimensions: []*cloudwatch.Dimension{
+					{
+						Name:  awsClient.String("DomainName"),
+						Value: cluster.DomainName,
+					},
+					{
+						Name:  awsClient.String("ClientId"),
+						Value: callerIdentityOutput.Account,
+					},
+				},
+			}
+
+			formulaValue, _, err := esm.cloudWatchCLient.GetMetric(&metricInput, metric)
+			if err != nil {
+				log.WithError(err).WithFields(log.Fields{
+					"cluster_id":  *cluster.ARN,
+					"metric_name": metric.Description,
+				}).Error("Could not get cloudwatch metric data")
+				continue
+			}
+
+			expression, err := expression.BoolExpression(formulaValue, metric.Constraint.Value, metric.Constraint.Operator)
+			if err != nil {
+				log.WithField("error", err).Error("could not parse expression")
+				continue
+			}
+
+			if expression {
+				hourlyClusterPrice := instancePrice*float64(*cluster.ElasticsearchClusterConfig.InstanceCount) + hourlyEBSVolumePrice
+				log.WithFields(log.Fields{
+					"metric_name":         metric.Description,
+					"Constraint_operator": metric.Constraint.Operator,
+					"Constraint_Value":    metric.Constraint.Value,
+					"formula_value":       formulaValue,
+					"cluster_id":          *cluster.ARN,
+					"node_type":           *cluster.ElasticsearchClusterConfig.InstanceType,
+					"region":              esm.region,
+				}).Info("ElasticSearch cluster detected as unutilized resource")
+
+				tags, err := esm.client.ListTags(&elasticsearch.ListTagsInput{
+					ARN: cluster.ARN,
+				})
+
+				tagsData := map[string]string{}
+				if err == nil {
```

Comment:
Do we want to add logs when `ListTags` returns error?

Reply:
added err check.

Reply:
why not remove this check? you already check if err is empty in line `210`